portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +32,50 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+    
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<Comment> comments = new ArrayList<>();
+
+    String commentLimitString = request.getParameter("limit");
+    int commentLimit = Integer.parseInt(commentLimitString);
+    if (commentLimit <= 0) commentLimit = 5;
+    for (Entity entity : results.asIterable()) {
```

Comment:
I know the instructions suggest using `.asIterable()` like this, but the `PreparedQuery` API [1] offers a way for you to limit the number of results returned from the datastore so you don't have to do all this counting yourself. You can use it like:

```
results.asIterable(FetchOptions.Builder.withLimit(commentLimit))
```

If you call `.asList(...)` instead then you can do this whole loop as:

```
results.asIterable(FetchOptions.Builder.withLimit(commentLimit)).forEach(entity -> { ... });
```

[1] https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/PreparedQuery

Reply:
Thank you. I knew there had to be a better solution than counting and breaking.